google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/features/InstanceApi.java

```diff
@@ -135,6 +135,17 @@ Operation deleteAccessConfigFromNic(@PathParam("instance") String instance,
    @GET
    @Path("/{instance}/serialPort")
    SerialPortOutput getSerialPortOutput(@PathParam("instance") String instance);
+   
+   /**
+    * Returns the specified instance's serial port output.
+    *
+    * @param instance the instance name.
+    * @return if successful, this method returns a SerialPortOutput containing the instance's serial output.
+    */
+   @Named("Instances:getSerialPortOutput")
+   @GET
+   @Path("/{instance}/serialPort")
+   SerialPortOutput getSerialPortOutput(@PathParam("instance") String instance, @QueryParam("port") int port);
```

Comment:
Add the corresponding [mock](https://github.com/jclouds/jclouds-labs-google/blob/1.9.x/google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/features/InstanceApiMockTest.java#L59-L66) and [live](https://github.com/jclouds/jclouds-labs-google/blob/1.9.x/google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/features/InstanceApiLiveTest.java#L160-L166) tests, to verify the behaviour when the query parameter is passed.